sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java

```diff
@@ -326,6 +337,83 @@ public static void reportFailedRPCMetrics(
     updateRpcLatencyMetric(c, method);
   }
 
+  /**
+   * Results of a batch of InsertAll RPCs. Used to update metrics with the {@code
+   * updateStreamingInsertsMetrics} method. Member variables should be thread-safe.
+   */
+  public static class StreamingInsertsResults {
+    final ConcurrentLinkedQueue<java.time.Duration> rpcLatencies = new ConcurrentLinkedQueue<>();
+    final ConcurrentLinkedQueue<String> rpcStatus = new ConcurrentLinkedQueue<>();
+    // Represents <Rpc Status, Number of Rows> for rows that are retried because the InsertAll RPC
+    // failed.
+    final ConcurrentLinkedQueue<KV<String, Integer>> retriedRowsByStatus =
+        new ConcurrentLinkedQueue<>();
+    final AtomicInteger successfulRowsCount = new AtomicInteger();
+    final AtomicInteger failedRowsCount = new AtomicInteger();
+    // Rows that were retried due to an internal BigQuery Error even though the InsertAll RPC
+    // succeeded.
+    final AtomicInteger internalRetriedRowsCount = new AtomicInteger();
+  }
+
+  /**
+   * Update the {@code RpcRequestsCount}, {@code RpcLatency}, and {@code RowsAppendedCount} metrics
+   * based on the results of StreamingInserts RPC calls.
+   *
+   * @param results Result of StreamingInsert RPC calls
+   * @param tableRef BigQuery table that was written to.
+   */
```

Comment:
how about making this an instance method w/in `StreamingInsertsResults`?

can be like 
```
public void updateMetrics(@Nullable TableReference tableRef) {

...//do operations on members instead of passed in StreamingInsertsResults.

}
```

this can be done even with AutoValue

Reply:
done.